Incompatibility with `--watch` and `declarationMap`

If you have `declarationMap: true` in your `tsconfig.json` (which I think is now implicitly set if you use `@tsconfig/ember`) and run `glint --watch`, you'll get the following error:

```
error TS5069: Option 'declarationMap' cannot be specified without specifying option 'declaration' or option 'composite'.
```

We most likely just need to account for it in the same way as we do other mutually-incompatible flags [here](https://github.com/typed-ember/glint/blob/cleaner-allowjs-handling/packages/core/src/cli/options.ts)

--- a/packages/core/src/cli/options.ts
+++ b/packages/core/src/cli/options.ts
@@ -21,6 +21,7 @@ export function determineOptionsToExtend(argv: {
   } else {
     options.noEmit = true;
     options.declaration = false;
+    options.declarationMap = false;
   }
 
   return options;

--- a/test-packages/ts-ember-app/tsconfig.json
+++ b/test-packages/ts-ember-app/tsconfig.json
@@ -17,6 +17,8 @@
     "noPropertyAccessFromIndexSignature": true,
     "noEmitOnError": false,
     "noEmit": true,
+    "declaration": true,
+    "declarationMap": true,
     "inlineSourceMap": true,
     "inlineSources": true,
     "baseUrl": ".",

